src/VisualStudio/Core/Def/Implementation/TaskList/ProjectExternalErrorReporter.cs

```diff
@@ -284,19 +290,47 @@ public int ClearErrors()
             int originalStartLine,
             int originalStartColumn,
             int originalEndLine,
-            int originalEndColumn)
+            int originalEndColumn,
+            DiagnosticAnalyzerInfoCache analyzerInfoCache)
         {
+            string title, description, category, helpLink;
+            DiagnosticSeverity defaultSeverity;
+            bool isEnabledByDefault;
+            ImmutableArray<string> customTags;
+
+            if (analyzerInfoCache != null && analyzerInfoCache.TryGetDescriptorForDiagnosticId(errorId, out var descriptor))
+            {
+                title = descriptor.Title.ToString(CultureInfo.CurrentCulture);
+                description = descriptor.Description.ToString(CultureInfo.CurrentCulture);
+                category = descriptor.Category;
+                defaultSeverity = descriptor.DefaultSeverity;
+                isEnabledByDefault = descriptor.IsEnabledByDefault;
+                customTags = descriptor.CustomTags.AsImmutableOrEmpty();
+                helpLink = descriptor.HelpLinkUri;
+            }
+            else
+            {
+                title = message;
+                description = message;
+                category = WellKnownDiagnosticTags.Build;
+                defaultSeverity = severity;
+                isEnabledByDefault = true;
+                customTags = IsCompilerDiagnostic(errorId) ? CompilerDiagnosticCustomTags : CustomTags;
+                helpLink = null;
```

Comment:
This may be improved a bit for compiler diagnostics. We can reuse this:

https://github.com/dotnet/roslyn/blob/5e8d77774d16ae50bc4f66deb41726c494283992/src/Compilers/CSharp/Portable/Errors/ErrorFacts.cs#L161-L164

It's a minor adjustment, so you can ignore if you think it doesn't worth it.